src/Uno.UI/UI/Xaml/FrameworkTemplatePool.cs

```diff
@@ -240,9 +240,10 @@ private void OnParentChanged(object instance, object key, DependencyObjectParent
 
 		internal static void PropagateOnTemplateReused(object instance)
 		{
-			if (instance is IFrameworkTemplatePoolAware a)
+			// If DataContext is not null, it means it has been explicitly set (not inherited). Resetting the view could push an invalid value through 2-way binding in this case.
```

Comment:
What happen if the DataContext is set directly on one of the children?

Reply:
`OnTemplateRecycled` is called individually for each view in the subtree, so it shouldn't matter.